File: reqquest-backend/reqquest-backend/Models/Request.cs
namespace reqquest_backend.Models;

public class Request
{
   public int Id { get; set; }
   public int CollectionId { get; set; }
   public Collection? Collection { get; set; }
   public string Name { get; set; } = string.Empty;
   public string Method { get; set; } = string.Empty;
   public DateTime CreatedAt { get; set; }
   public DateTime UpdatedAt { get; set; }
}

File: reqquest-backend/entities/request.go
package entities

import "time"

type Method string

const (
	Get    Method = "get"
	Post   Method = "post"
	Patch  Method = "patch"
	Put    Method = "put"
	Delete Method = "delete"
)

type Request struct {
	ID           int       `json:"id"`
	Name         string    `json:"name"`
	Method       Method    `json:"method"`
	CollectionId int       `json:"collectionId"` //Foreign Key Reference
	CreatedAt    time.Time `json:"createdAt"`
}
